services/path_service.py

```
@@ -197,35 +214,69 @@ def construct_destination_from_components(
         else:
             # Movie or series folder
             dest_path = os.path.join(dest_base, folder_name)
-        
+
+        # SECURITY: Double-check resolved path stays within bounds
+        assert_path_within_bounds(dest_path, [dest_base])
+
         return dest_path
     
-    def validate_destination_path(self, dest_path: str) -> bool:
+    def validate_destination_path(self, dest_path: str, media_type: str = None) -> bool:
```

Ruff (0.15.10)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)